accounts-db/src/accounts_db.rs

```diff
@@ -8853,15 +8853,24 @@ impl AccountsDb {
         let mut amount_to_top_off_rent = 0;
         let mut stored_size_alive = 0;
 
+        let mut all_accounts_are_zero_lamports = true;
+
         let (dirty_pubkeys, insert_time_us, mut generate_index_results) = {
             let mut items_local = Vec::default();
             storage.accounts.scan_index(|info| {
                 stored_size_alive += info.stored_size_aligned;
                 if info.index_info.lamports > 0 {
                     accounts_data_len += info.index_info.data_len;
+                    all_accounts_are_zero_lamports = false;
                 }
                 items_local.push(info.index_info);
             });
+            if all_accounts_are_zero_lamports {
+                // this whole slot can likely be marked dead and dropped. Clean has to determine that. There could be an older non-zero account for any of these zero lamport accounts.
+                self.dirty_stores.insert(slot, Arc::clone(storage));
+                self.accounts_index.add_uncleaned_roots([slot].into_iter());
+                self.shrink_candidate_slots.lock().unwrap().insert(slot);
```

Comment:
I don't know for sure we should do this. we may want to let clean calculate whether we can shrink or not. I suspect we don't want to add to shrink canddiate.